eml/api/api.js

```diff
@@ -179,3 +179,68 @@ export const ifSubscribed = async (userId, courseId) => {
     }
   }
 };
+
+//CREATED BY VIDEOSTREAM TEAM
+/**
+ *
+ * @param {* name of video in bucketRoute, should be the same as lecture id} fileId
+ * @param {* use 360p, 720p or 1080p - standard is 360p} resolution
+ */
+export const getVideoDownloadUrl = (fileName, resolution) => {
+  let usableResolution = "360x640";
+
+  switch (resolution) {
+    case "180p":
+      usableResolution = "180x320";
+      break;
+    case "360p":
+      usableResolution = "360x640";
+      break;
+    case "720p":
+      usableResolution = "720x1280";
+      break;
+    case "1080p":
+      usableResolution = "1080x1920";
+      break;
+    default:
+      usableResolution = "360x640";
+      break;
+  }
```

Comment:
**Suggestion: Use an enum for resolution values**

 Current implementation uses a switch statement to handle different resolution inputs. However, this approach can be prone to errors, especially if an invalid resolution is passed. I would suggested to use an enum to define a set of fixed resolution. Enum could look something along:
 
```js
enum Resolution {
    P180 = "180x320",
    P360 = "360x640",
    P720 = "720x1280",
    P1080 = "1080x1920"
}
```

Reply:
We are working on another solution for a microservice rework, where we will change it altogether.